src/LibertyApp/Models/ConnectionSpeed.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LibertyApp.Models;
+
+public class ConnectionSpeed
+{
+    private readonly NetworkInterface[] _networkInterfaces;
+    private Dictionary<NetworkInterface, long> _lastReceivedBytes;
+    private Dictionary<NetworkInterface, long> _lastSentBytes;
+
+    public ConnectionSpeed()
+    {
+        _networkInterfaces = NetworkInterface.GetAllNetworkInterfaces();
```

Comment:
Do you want to watch all network interfaces at the same time? Is it possible to track only our vpn connection through the WAN Miniport?

Reply:
I thought that when we are connected to VPN all our traffic from all interfaces goes through it